pennylane/tape/qnode.py

```diff
@@ -546,6 +567,10 @@ def to_tf(self, dtype=None):
             from pennylane.tape.interfaces.tf import TFInterface
 
             self.interface = "tf"
+            self._tape, self.interface, diff_method, self.device = self.get_tape(

```

Comment:
Maybe worth adding a comment explaining this bit? Hasn't `get_tape` already been called upon instantiation?

Reply:
Good point! The reason I added it here was to support the `qnn` module, where we make use of the fact that a QNode with an existing interface can be converted to another interface. Perhaps the following is less redundant?

```python
if self.interface != "tf":
    # Interface is changing, need to re-validate the tape class.
    self._tape, self.interface, diff_method, self.device = self.get_tape(
        self.original_device, self.interface, self.diff_method
    )
self.interface = "tf"
```

Reply:
Have made this change :)